CONTRIBUTING.md

```diff
@@ -1,24 +1,63 @@
+**Before you contribute**
+
+* Check out the roadmap: This will give you an overview of the project's direction and help you identify areas where your contributions can have the greatest impact.
+* Take a look into the existing issues: Check if your issue or challenge has already been discussed or reported. This helps avoid duplication of effort.
+* Learn about the code: The Lumigator project consists of several packages, each with its own pyproject.toml file. Understanding the project structure will make it easier to contribute effectively. The lumigator project contains several packages: the `lumigator/python/mzai/backend` package that powers the lumigator server functionality, the `lumigator/python/mzai/schemas` package containing the formal schemas for communication with the server through the REST API, and the `lumigator/python/mzai/sdk` package abstracting the REST API for Python applications. Each package holds its own `pyproject.toml` definition.
+
+### **Disclaimer**
+
+We appreciate all contributions and are eager to review them. However, please understand that we may not be able to accept every contribution due to factors such as team capacity, project priorities, and the overall scope of the project.
+
+We will carefully consider all submissions and provide feedback whenever possible. If your contribution is not accepted, please don't be discouraged! We value your input and appreciate you taking the time to contribute.
+
+## **How Can I Contribute?**
+
+**Reporting Bugs**
+
+If you find an error while running Lumigator, we would appreciate you opening an [issue](https://github.com/mozilla-ai/lumigator/issues) to report it. Follow the corresponding template in Github providing a description of the bug, a detailed description on how to reproduce the reported behavior, a detailed description on what you would expect to happen and your system info. 
+
+**Suggesting features**
+
+If you have ideas for improvements or new features, feel free to open a new discussion or issue to share them. We appreciate well documented ideas or  suggestions. Clearly explain the problem you're trying to solve, the proposed solution, and any relevant diagrams or examples. You can also have at look to our ROADMAP to check if your proposal is aligned with the defined direction of the Lumigator.
+
+We initiate a [discussion](https://github.com/mozilla-ai/lumigator/discussions) on GitHub to refine the implementation details. Once clarified, the feature can be implemented by either us or the reporter." make an upgrade of our contributing file combining the best from all the inputs I gave you and propose me things that can be useful to add.
+
+**Writing Code**
+
+If you don’t know where to start, there is a special [Good First Issue](https://github.com/mozilla-ai/lumigator/contribute). It will give you a list of open issues that are beginner friendly and help you start contributing to our project. 
```

Comment:
```suggestion
If you don’t know where to start, there is a special [Good First Issue](https://github.com/mozilla-ai/lumigator/contribute) page. It will give you a list of open issues that are beginner friendly and help you start contributing to our project. 
```